etrago/cluster/networkclustering.py

```diff
@@ -781,23 +781,81 @@ def fetch():
         busmap = fetch()
 
     return busmap
+    
+def delete_ehv_buses_no_lines(network):
+    """
+    When there are AC buses totally isolated, this function deletes them in order
+    to make possible the creation of busmaps based on electrical connections
+    and other purposes. Additionally, it throws a warning to inform the user
+    in case that any correction should be done.
 
+    Parameters
+    ----------
+    network : pypsa.network
 
+    Returns
+    -------
+    None
+    """
+    lines = network.lines
+    buses_ac = network.buses[(network.buses.carrier == "AC") &
+                             (network.buses.country == "DE")]
+    buses_in_lines = set(list(lines.bus0) + list(lines.bus1))
+    buses_ac["with_line"] = buses_ac.index.isin(buses_in_lines)
+    buses_ac["with_load"] = buses_ac.index.isin(network.loads.bus)
+    buses_in_links = list(network.links.bus0)+list(network.links.bus1)
+    buses_ac["with_link"] = buses_ac.index.isin(buses_in_links)
+    buses_ac["with_gen"] = buses_ac.index.isin(network.generators.bus)
+
+    delete_buses = buses_ac[(buses_ac["with_line"] == False) &
+                            (buses_ac["with_load"] == False) &
+                            (buses_ac["with_link"] == False) &
+                            (buses_ac["with_gen"] == False)].index
+
+    if len(delete_buses):
+        logger.info(f"""
+                    
+                    ----------------------- WARNING ---------------------------
+                    THE FOLLOWING BUSES WERE DELETED BECAUSE THEY WERE ISOLATED:
+                        {delete_buses.to_list()}.
+                    IT IS POTENTIALLY A SIGN OF A PROBLEM IN THE DATASET
+                    ----------------------- WARNING ---------------------------
+                    
+                    """)
+                    
+    network.buses = network.buses.drop(delete_buses)
```

Comment:
These lines are working, but in general I would recommend to use the build-in pypsa functions to drop components (see e.g. https://github.com/openego/eTraGo/blob/ad10466425386b2aa42a6f78c4f03d562188783d/etrago/tools/utilities.py#L1712-L1715). When you use them, e.g. time series are dropped automatically.

Reply:
The function mremove is used now to delete elements.